benchmarks/cpp_micro_benchmarks.py

```diff
@@ -114,47 +88,23 @@ class RecordCppMicroBenchmarks(_benchmark.Benchmark):
     description = "Run the Arrow C++ micro benchmarks."
     iterations = 1
     flags = {"language": "C++"}
+    adapter = None
 
-    def run(self, **kwargs):
-        with tempfile.NamedTemporaryFile(delete=False) as result_file:
-            run_command = get_run_command(result_file.name, kwargs)
-            self.execute_command(run_command)
-            results = json.load(result_file)
-            for suite in results["suites"]:
-                self.conbench.mark_new_batch()
-                for result in suite["benchmarks"]:
-                    yield self._record_result(suite, result, kwargs)
-
-    def _record_result(self, suite, result, options):
-        info, context = {}, {"benchmark_language": "C++"}
-        tags = _parse_benchmark_name(result["name"])
-        name = tags.pop("name")
-        tags["suite"] = suite["name"]
-        tags["source"] = self.name
-
-        values = self._get_values(result)
-
-        return self.record(
-            values,
-            tags,
-            info,
-            context,
-            options=options,
-            output=result,
-            name=name,
+    def __init__(self):
+        tags, info, context = self._get_tags_info_context(case=None, extra_tags={})
+
+        self.adapter = ArcheryAdapter(
+            result_fields_override={"github": self.github_info},
```

Comment:
We might want to have a comment here talking about why we're relying on github_info from _benchmark.py